packages/console/src/pages/TenantSettings/TenantMembers/EditMemberModal/index.tsx

```diff
@@ -0,0 +1,87 @@
+import { TenantRole } from '@logto/schemas';
+import { getUserDisplayName } from '@logto/shared/universal';
+import { useContext, useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import ReactModal from 'react-modal';
+import { z } from 'zod';
+
+import { useAuthedCloudApi } from '@/cloud/hooks/use-cloud-api';
+import { type TenantMemberResponse } from '@/cloud/types/router';
+import { TenantsContext } from '@/contexts/TenantsProvider';
+import Button from '@/ds-components/Button';
+import FormField from '@/ds-components/FormField';
+import ModalLayout from '@/ds-components/ModalLayout';
+import Select from '@/ds-components/Select';
+import * as modalStyles from '@/scss/modal.module.scss';
+
+type Props = {
+  user: TenantMemberResponse;
+  isOpen: boolean;
+  onClose: () => void;
+};
+
+function EditMemberModal({ user, isOpen, onClose }: Props) {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console.tenant_members' });
+  const { currentTenantId } = useContext(TenantsContext);
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [role, setRole] = useState(TenantRole.Member);
+  const cloudApi = useAuthedCloudApi();
+
+  const roleOptions = useMemo(
+    () => [
+      { value: TenantRole.Admin, title: t('admin') },
+      { value: TenantRole.Member, title: t('member') },
+    ],
+    [t]
+  );
+
+  const onSubmit = async () => {
```

Comment:
wrap up using a useCallback hook?

Reply:
I think it's not necessary in this case. `useCallback` should only be needed when it causes unexpected re-render of the component. IIRC, The official document does not recommend wrapping every function in `useCallback` hook.

Reply:
Will the onSubmit function redefined on every render cycle?

Reply:
Yes, but this one is light-weighted and it doesn't matter. useCallback also has performance costs, so my principle is don't use unless it's necessary.

Here's a blog post about it: https://kentcdodds.com/blog/usememo-and-usecallback